Foundation/GTMNSThread+BlocksTest.m

```diff
@@ -316,12 +316,33 @@ - (void)testStopFromOtherThread {
 - (void)testPThreadName {
   NSString *testName = @"InigoMontoya";
   [workerThread_ setName:testName];
+  // There is actually a race between setting the thread name in NSThread and it
+  // being visible in the thread itself. A thread's name can only be set from
+  // inside the thread (pthread_setname) so we are assuming that NSThread has
+  // some internal wiring to kick the runloop and update the value for the
+  // thread on the thread itself.
+  // If we run our "check" block too soon, we will check the value on the
+  // thread before it's been set.
+  // Therefore we loop over it with some sleep to give the thread a chance
+  // to update it's internals.
   [workerThread_ gtm_performWaitingUntilDone:YES block:^{
     XCTAssertEqualObjects([workerThread_ name], testName);
```

Comment:
Actually, shouldn't this have come out?  Otherwise the it can still fail even though the code below lets it sleep and eventually take effect?